Lib/fontTools/ttLib/tables/TupleVariation.py

```diff
@@ -54,10 +54,7 @@ def hasImpact(self):
 		If the result is False, the TupleVariation can be omitted from the font
 		without making any visible difference.
 		"""
-		for c in self.coordinates:
-			if c is not None:
-				return True
-		return False
+		return any(c is not None for c in self.coordinates)
```

Comment:
Should we check for 0s as well? If all coordinates are 0, then its not going to have an impact.

Reply:
no, that would change the meaning of that method, which pre-existed (I simply made it more idiomatic by using `any`). The optimize method will make sure that if a tuple has all zeros, it will be optimized away, cf. https://github.com/fonttools/fonttools/blob/f183139f5abe3a4fb64e96f7ab9ad078db73d8a3/Lib/fontTools/varLib/iup.py#L226-L228